analytics-event-publisher-archetype/src/main/resources/archetype-resources/src/main/java/__carbon_analytics_publisher_name__EventAdapterFactory.java

```diff
@@ -0,0 +1,103 @@
+/*
+*  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package ${package};
+
+import org.wso2.carbon.event.output.adapter.core.OutputEventAdapter;
+import org.wso2.carbon.event.output.adapter.core.OutputEventAdapterConfiguration;
+import org.wso2.carbon.event.output.adapter.core.OutputEventAdapterFactory;
+import org.wso2.carbon.event.output.adapter.core.Property;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The ${carbon_analytics_publisher_name} event adapter factory class to create
+ * an ${carbon_analytics_publisher_name} output adapter
+ */
+public class ${carbon_analytics_publisher_name}EventAdapterFactory extends OutputEventAdapterFactory {
+
+    /**
+     * Here type needs to be specified,
+     * this string will be displayed in the publisher interface in the adapter type drop down list.
+     *
+     * @return
+     */
+    @Override
+    public String getType() {
+        return null;
+    }
+
+    /**
+     * Specify supported message formats for the created publisher type.
+     *
+     * @return
+     */
+    @Override
+    public List<String> getSupportedMessageFormats() {
+        return null;
```

Comment:
There is no special reason to leave it as null.
Fixed in https://github.com/wso2-extensions/archetypes/pull/10